src/main/java/Interpreter.java

```diff
@@ -1,87 +1,88 @@
-            } else if (operators.contains("+")) {
-                index = operators.indexOf("+");
-            } else if (operators.contains("-")) {
-                index = operators.indexOf("-");
-            }
-            return interpret(choseAction(operands, operators, index));
+            return interpret(choseAction(operands, operators, getIndex(operators)));
         } else return inputExpression;
 
     }
 
-    private String choseAction(List<String> operands, List<String> operators, int index) {
-        String replacmentExpression = "";
-        switch (operators.get(index)) {
-            case "*":
-                replacmentExpression = calculator.multiplyNumbers(operands, operators, index);
-                break;
-            case "/":
-                replacmentExpression = calculator.divideNumbers(operands, operators, index);
-                break;
-            case "+":
-                replacmentExpression = calculator.addNumbers(operands, operators, index);
-                break;
-            case "-":
-                replacmentExpression = calculator.subNumbers(operands, operators, index);
-                break;
+    private int getIndex(List<String> operators) {
+        int index = 0;
+        if (operators.contains("/") && operators.contains("*")) {
+            index = Math.min(operators.indexOf("*"), operators.indexOf("/"));
+        } else if (operators.contains("/")) {
+            index = operators.indexOf("/");
+        } else if (operators.contains("*")) {
+            index = operators.indexOf("*");
+        } else if (operators.contains("+") && operators.contains("-") && operators.indexOf("-") != 0) {
+            index = Math.min(operators.indexOf("+"), operators.indexOf("-"));
+        } else if (operators.contains("+")) {
+            index = operators.indexOf("+");
+        } else if (operators.contains("-")) {
+            index = operators.indexOf("-") > 0 ? operators.indexOf("-") : 1;
         }
-        return modifyExpression(operands, operators, index, replacmentExpression);
+        return index;
     }
 
-    private String modifyExpression(List<String> operands, List<String> operators, int index, String replacmentExpression) {
-        operands.set(index, replacmentExpression);
+    private String choseAction(List<String> operands, List<String> operators, int index) {
+        String replacementExpression = switch (operators.get(index)) {
+            case "*" -> calculator.multiplyNumbers(operands, index);
+            case "/" -> calculator.divideNumbers(operands, index);
+            case "+" -> calculator.addNumbers(operands, index);
+            case "-" -> calculator.subNumbers(operands, index);
+            default -> "";
+        };
+        return modifyExpression(operands, operators, index, replacementExpression);
+    }
+
+    private String modifyExpression(List<String> operands, List<String> operators, int index, String replacementExpression) {
```

Comment:
wouldn't it be simpler to just replace calculated part in an original expression rather than building it from scratch?